src/package.ts

```diff
@@ -0,0 +1,57 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+export class SPMPackage {
+	private constructor(
+        readonly folder: string,
+        public contents: any
+    ) {
+        this.setContextKeys()
+    }
+
+    public static async create(folder: string): Promise<SPMPackage> {
```

Comment:
Same question here: Do we really need the factory method?

Reply:
Again it is an async function and you cannot have an async constructor

Reply:
Got it!